rust/agama-server/src/manager/web.rs

```diff
@@ -173,15 +173,21 @@ async fn install_action(State(state): State<ManagerState<'_>>) -> Result<(), Err
 /// Executes the post installation tasks (e.g., rebooting the system).
 #[utoipa::path(
     post,
-    path = "/install",
+    path = "/finish",
     context_path = "/api/manager",
     responses(
-      (status = 200, description = "The installation tasks are executed.")
+      (status = 200, description = "The installation tasks are executed.", body = Option<FinishMethod>)
     )
 )]
-async fn finish_action(State(state): State<ManagerState<'_>>) -> Result<(), Error> {
-    state.manager.finish().await?;
-    Ok(())
+async fn finish_action(
+    State(state): State<ManagerState<'_>>,
+    method: Option<Json<FinishMethod>>,
+) -> Result<Json<bool>, Error> {
+    if let Some(Json(method)) = method {
```

Comment:
Just a suggestion:

```rust
let method = method.unwrap_or_default();
state.manager.finish(method).await?; // you might need to adapt the `finish` function.
```

Reply:
I was not sure where to set it as mandatory. Thanks ;)

Reply:
As discussed offline, the problem is that `Json` and `match` could be the best option:

```rust
    let method = match method {
        Some(Json(method)) => method,
        _ => FinishMethod::default(),
    };

    Ok(Json(state.manager.finish(method).await?))
```